tasks/eva_2126/check_mapping_weight.py

```diff
@@ -0,0 +1,71 @@
+#!/usr/bin/env python
+from argparse import ArgumentParser
+from urllib.parse import quote_plus
+
+import pymongo
+
+
+def get_mongo_connection_handle(host, port=27017, username=None, password=None, authentication_database="admin", **kwargs) -> pymongo.MongoClient:
+    mongo_connection_uri = "mongodb://"
+    if username and password:
+        mongo_connection_uri += '%s:%s@' % (username, quote_plus(password))
+    mongo_connection_uri += '%s:%s/%s' % (host, port, authentication_database)
+    return pymongo.MongoClient(mongo_connection_uri)
+
+
+def check_mapping_weight(mongo_host, database_name, username, password, reference_accession):
+    """
+    Connect to mongodb and retrieve all variants the should be updated, Check their key and update them in bulk.
+    """
+    with get_mongo_connection_handle(mongo_host, username=username, password=password) as accessioning_mongo_handle:
+        dbsnp_cve_collection = accessioning_mongo_handle[database_name]["dbsnpClusteredVariantEntity"]
+        cursor = dbsnp_cve_collection.find({'mapWeight': {'$gt': 1}, 'asm': reference_accession})
+        count_variants = 0
+        list_variant_erroneous = []
+        erroneous_recovered1 = 0
+        erroneous_recovered2 = 0
+        for variant in cursor:
+            count = dbsnp_cve_collection.count({'accession': variant['accession']})
+            count_variants += 1
+            if count < 2:
+                list_variant_erroneous.append(variant)
+
+        dbsnp_cve_op_collection = accessioning_mongo_handle[database_name]["dbsnpClusteredVariantOperationEntity"]
+        dbsnp_sve_op_collection = accessioning_mongo_handle[database_name]["dbsnpSubmittedVariantOperationEntity"]
+
+        for variant in list_variant_erroneous:
+            operation1 = list(dbsnp_cve_op_collection.find({'accession': variant['accession']}))
+            operation2 = list(dbsnp_sve_op_collection.find({'inactiveObjects.rs': variant['accession']}))
+
+            if operation1 and operation1[0]['eventType'] == 'MERGED':
+                erroneous_recovered1 += 1
+            elif operation2 and operation2[0]['eventType'] == 'UPDATED':
+                erroneous_recovered2 += 1
+            else:
+                print(
+                    "rs%s Should have more than 1 variant because mapWeight = %s" % (
+                    str(variant['accession']), variant['mapWeight'])
+                )
+                print("Found %s operation for clustered variants" % len(operation1))
+                print("Found %s operation for submitted variants" % len(operation2))
+
+        print("Checked %s variants" % count_variants)
```

Comment:
Maybe it is important to specify when we are referring to submitted or clustered variants to avoid confusion

Reply:
Changed it.